### ISSUE TITLE
Unable to set setpoint and thermostat states

### ISSUE THREAD

**Issue by JohnvandeVrugt** [costastf/toonapilib#23]
Hello,

First of all I like the work you did on the toon api!
Very nice.

I have created a plug-in for domiticz so we can have a working toon integration once more.
It's called toonapilib4domoticz. It's on GitHub as well.

However I seem to have issues with setting the temperature and states. They do not seem to be working. Reading stuff works like a charm.

Hope you can help me on this matter.
I follow your examples to set them. 

Thanks,
John

**Comment by costastf** [comment]
Hi @JohnvandeVrugt , I am afraid that I won't be able to look into this before tomorrow evening. I hope that's ok with you time wise.

**Comment by costastf** [comment]
Shall we move this over to your code issue tracker? Unless you have used the library by itself and that does not work in which case please ping me to re open this.

**Comment by JohnvandeVrugt** [comment]
Setting the temperature setpoint does not work in both my plugin and a simple test script.
I'm using toonapilib version 3.0.9 with Python v3.5. (raspbian)

The simple test script yields:
`from toonapilib import Toon

eneco_username = '***'
eneco_password = '***'
consumer_key='***'
consumer_secret='***'

toon=Toon(eneco_username, eneco_password, consumer_key, consumer_secret)
toon.thermostat = 18.5
`
The set point DOES change when I disable the program. 
Do I need to proceed this simple call of use a different api call to override the setpoint while a program is active?

**Comment by costastf** [comment]
AHA! That is actually a great point! I just tried and true enough if the program is on, the temperature does not change at least not all the time. That means that there is a race condition with another undocumented thing. The documentation states that one has to set 
```
data["currentSetpoint"] = target
```
but actually one has to also set 
```
data["activeState"] = -1
``` 
Now apparently if the program is on there is something else that needs to be set. I need to look into that. Reopening for now, but I won't really have time before the weekend for this I am afraid. You can always reach out to the toon developer people and report this as a bug. I did when I figured the above up but they did not update their documentation anyhow.

**Comment by JohnvandeVrugt** [comment]
Appreciate the rapid response!
Surely it can wait for a couple of days. 

If I figure something out, I won't hasitate to share my findings.
I think I have seen similar things when activating programs/states.

**Comment by costastf** [comment]
Sure enough, the api required 
```
data["programState"] = -1
``` 
set also for the temperature to be set. This is actually a great catch as I had people reporting this issue which I could not reproduce because I was not using programs until a while back. Thanks! This should be fixed in v3.0.10.
